docs/advanced-guide/grpc/page.md

```diff
@@ -1,125 +1,149 @@
-```protobuf
-// Indicates the protocol buffer version that is being used
-syntax = "proto3";
-// Indicates the go package where the generated file will be produced
-option go_package = "";
-
-service CustomerService {
-  // ...
-}
-```
-Inside the service one can define all the `rpc` methods, specifying the request and responses types.
-```protobuf
-service CustomerService {
-  // GetCustomer is a rpc method to get customer data using specific filters
-  rpc GetCustomer(CustomerFilter) returns(CustomerData) {}
-}
+
+**1. Protocol Buffer Compiler (`protoc`) Installation:**
+
+- **Linux (using `apt` or `apt-get`):**
+```bash
+  sudo apt install -y protobuf-compiler
+  protoc --version  # Ensure compiler version is 3+
+  ```
+
+- **macOS (using Homebrew):**
+
+```bash
+  brew install protobuf
+  protoc --version  # Ensure compiler version is 3+
 ```
-The `CustomerFilter` and `CustomerData` are two types of messages that will be exchanged between server
-and client. Users must define those for protocol buffer to serialize them when making a remote procedure call.
-```protobuf
-syntax = "proto3";
 
+**2. Go Plugins for Protocol Compiler:**
+
+a. Install protocol compiler plugins for Go:
+
+   ```bash
+     go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28
+     go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2
+   ```
+
+b. Update `PATH` for `protoc` to locate the plugins:
+
+   ```bash
+     export PATH="$PATH:$(go env GOPATH)/bin"
+   ```
+
+## Creating Protocol Buffers
+
+For a detailed guide, refer to the official gRPC documentation's tutorial: {% new-tab-link title="Tutorial" href="https://grpc.io/docs/languages/go/basics/" /%} at official gRPC docs.
+
+**1. Define Your Service and RPC Methods:**
+
+Create a `.proto` file (e.g., `customer.proto`) to define your service and the RPC methods it provides:
+
+   ```protobuf
+   // Indicates the protocol buffer version that is being used
+   syntax = "proto3";
+   // Indicates the go package where the generated file will be produced
+   // assuming proto file is defined in package with path github.com/grpc/customer
+   option go_package = "github.com/grpc/customer";
+
+   service CustomerService {
+       rpc GetCustomer (CustomerFilter) returns (CustomerData) {}
+   }
+   ```
+
+**2. Specify Request and Response Types:**
+
+For example: The CustomerFilter and CustomerData are two types of messages that will be exchanged between server and client. Users must define those for protocol buffer to serialize them when making a remote procedure call.
+
+```go
 message CustomerFilter {
-  int64 id = 1;
-  string name = 2;
-  // other fields that can be passed
+int64 id = 1;
+string name = 2;
+// other fields that can be passed
 }
 
 message CustomerData {
-  int64 id = 1;
-  string name = 2;
-  string address = 3;
-  // other customer related fields
+int64 id = 1;
+string name = 2;
+string address = 3;
+// other customer related fields
 }
 ```
```

Comment:
This is not valid Go code, I'm unsure about the purpose and what it is

So maybe you could remove the `go` identifier, or find a suitable fenced doc block identifier